community/kernel/src/main/java/org/neo4j/kernel/impl/query/Neo4jTransactionalContext.java

```diff
@@ -212,13 +204,18 @@ public TransactionalContext provideContext()
         {
             InternalTransaction transaction = graph.beginTransaction( transactionType, mode );
             Statement statement = txBridge.get();
+            statement.queryRegistration().registerExecutingQuery( executingQuery );
             return new Neo4jTransactionalContext(
-                graph,
-                transaction,
-                statement,
-                executingQuery.queryText(),
-                executingQuery.queryParameters(),
-                locker
+                    graph,
```

Comment:
We are using several private field which are more easily accessed here. Renamed the method instead to improved clarity.